### PR TITLE
Fix return value of uninstall.IsSupportedForVersion

### PR DESCRIPTION
**Pull request by sridhargaddam**
Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

File: controllers/uninstall/uninstall.go [hunk 1]
```diff
@@ -325,5 +325,5 @@ func findPodsBySelector(ctx context.Context, clnt client.Reader, namespace strin
 
 func IsSupportedForVersion(version string) bool {
 	semVersion, _ := semver.NewVersion(version)
-	return !(semVersion != nil && !semVersion.LessThan(*minComponentUninstallVersion))
+	return semVersion != nil && !semVersion.LessThan(*minComponentUninstallVersion)
```
Comment (sridhargaddam) [inline controllers/uninstall/uninstall.go]: @tpantelis This PR tries to match the return value with the Function name. 

At the same time, I see that when using KIND deployments, the `submariner` version will be `devel`, so un-installation is not triggered. What were your thoughts about it?

Comment (tpantelis) [reply controllers/uninstall/uninstall.go]: It should trigger for `devel` - this was inadvertently broken by https://github.com/submariner-io/submariner-operator/pull/1843/.  The check should be:

```suggestion
	return semVersion == nil || !semVersion.LessThan(*minComponentUninstallVersion)
```


We should have a test case to cover `devel`. I suggest we change the set up for [this test](https://github.com/submariner-io/submariner-operator/blob/9ed748bcd4db5f639f2c056e875fc9dec882cbf6/controllers/submariner/submariner_controller_test.go#L319) to add `t.submariner.Spec.Version = "devel"`.

Comment (sridhargaddam) [reply controllers/uninstall/uninstall.go]: Done, thanks @tpantelis

File: controllers/uninstall/uninstall.go [hunk 2]
```diff
@@ -325,5 +325,5 @@ func findPodsBySelector(ctx context.Context, clnt client.Reader, namespace strin
 
 func IsSupportedForVersion(version string) bool {
 	semVersion, _ := semver.NewVersion(version)
-	return !(semVersion != nil && !semVersion.LessThan(*minComponentUninstallVersion))
+	return semVersion == nil || !semVersion.LessThan(*minComponentUninstallVersion)
```
Comment (skitt) [inline controllers/uninstall/uninstall.go:328]: Does it make sense to return `true` when we don’t know the version?

Comment (skitt) [reply controllers/uninstall/uninstall.go:328]: Oh, I hadn’t seen the previous comments.

**Review by skitt — APPROVED** [review]

**Review by Jaanki — APPROVED** [review]

**Pull request merged**